tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -142,6 +142,22 @@ def kill_all(self):
         # Now kill the user apps
         self.marionette.switch_to_frame()
         self.marionette.execute_async_script("GaiaApps.killAll();")
+    
+    def install(self, manifest_url):
+        self.marionette.switch_to_frame()
+        data_layer = GaiaData(self.marionette)
+        data_layer.set_bool_pref('dom.mozApps.auto_confirm_install', True)
+        result = self.marionette.execute_async_script('GaiaApps.installWithManifestURL("%s")' % manifest_url)
+        assert (result is True), 'Failed to install app: %s' % result
+        data_layer.set_bool_pref('dom.mozApps.auto_confirm_install', False)
+
+    def uninstall(self, manifest_url):
+        self.marionette.switch_to_frame()
+        data_layer = GaiaData(self.marionette)
+        data_layer.set_bool_pref('dom.mozApps.auto_confirm_uninstall', True)
+        result = self.marionette.execute_async_script('GaiaApps.uninstallWithName("%s")' % manifest_url)
```

Comment:
This has to become 'GaiaApps.uninstallWithmanifestURL.
https://marketplace.firefox.com/app/9b628257-c986-47aa-bdb9-a654ec355108/manifest.webapp

You can't have 2 def uninstall's in here, you'll have to delete the one with the name.
That means you can't use name anymore for uninstall and you'll have to fix all those instances.
Regarding test_marketplace_packaged_app.py, the app to uninstall has manifest_url https://marketplace.firefox.com/app/9b628257-c986-47aa-bdb9-a654ec355108/manifest.webapp , you can use that.